src/clj_http/util.clj

```diff
@@ -44,7 +44,10 @@
   (when b
     (cond
       (instance? InputStream b)
-      (GZIPInputStream. b)
+      (try 
+        (GZIPInputStream. b)
+        (catch java.io.EOFException e
+          nil))
```

Comment:
I couldn't reproduce this particular scenario locally :/ As mentioned above, I think the work to consider migration to Github Actions is out of scope for this PR and could be tackled in a separate PR.

Reply:
Hmm, I'll see if it is still happening.